src/base/NekRSProblem.C

```diff
@@ -694,20 +851,23 @@ NekRSProblem::syncSolutions(ExternalProblem::Direction direction)
       if (_volume && _has_heat_source)
         sendVolumeHeatSourceToNek();
 
-      // copy the boundary heat flux and/or volume heat source in the scratch space to device
-      nekrs::copyScratchToDevice(_minimum_scratch_size_for_coupling);
 
-      if (_moving_mesh)
+      if (nekrs::hasUserMeshSolver() && _volume)
       {
-        if (_volume)
-        {
-          sendVolumeDeformationToNek();
-          nekrs::copyDeformationToDevice();
-        }
-
-        // no boundary-based mesh movement available in nekRS yet
+        sendVolumeDeformationToNek();
+      }
+      else if (nekrs::hasElasticitySolver())
+      {
+        sendBoundaryDeformationToNek();
       }
 
+      // copy the boundary heat flux and/or volume heat source in the scratch space to device
+      nekrs::copyScratchToDevice(_minimum_scratch_size_for_coupling);
+      if (nekrs::hasMovingMesh())
```

Comment:
Should this line actually be `if (nekrs::hasUserMeshSolver())`?

Reply:
```
      if (nekrs::hasMovingMesh())
      {
      // update geometric factors and the mesh on the device
        nekrs::copyDeformationToDevice();
      }
```
We decided to do this every time regardless of whether we are using the `user` mesh solver (with a volume mirror) or the `elasticity` solver (works with either volume or boundary mirrors). This is because it seemed to us that the nekRS team was making changes to their mesh solvers across different versions and sometimes this was necessary with one solver, and sometimes with another. So for consistency, we decided to leave this in for all moving mesh solvers since this update is relatively cheap and ensures that future versions of Cardinal don't break due to changes in nekRS.

Reply:
Got it, thanks for the detailed explanation. Would you please add a comment into the code to this effect, in case future developers have the same question?

Reply:
Done, thanks!